File: clojure/word-count/src/word_count.clj
(ns word-count)

(defn words [phrase]
  (clojure.string/split phrase #"\s+"))

(defn clean[phrase]
  (clojure.string/replace
    phrase
    #"[^A-Za-z0-9 ]"
    ""))

(defn word-count [phrase]
  (->> phrase
    (clean)
    (words)
      (map clojure.string/lower-case)
        (frequencies)))

File: python/word-count/wordcount.py
from collections import Counter
import re

def word_count(string):
    formatted_string = format_string(string)
    return Counter(formatted_string.split())

def format_string(string):
    return re.sub(r'[^0-9a-zA-Z \_\\n\\t]+', " ", string).replace("_", " ").lower()
